patientsearch/api.py

```diff
@@ -203,10 +204,19 @@ def external_search(resource_type, methods=["GET"]):
 @api_blueprint.route('/logout', methods=["GET"])
 def logout(methods=["GET"]):
     # TODO: is there a PHI safe 'id' for the user (in place of email)?
-    user_id = oidc.user_getfield('email')
-    current_app.logger.info(
-        "logout on request",
-        extra={'tags': ['logout'], 'user_id': user_id})
-    terminate_session()
+    try:
+        user_id = oidc.user_getfield('email')
+        current_app.logger.info(
+            "logout on request",
+            extra={'tags': ['logout'], 'user_id': user_id})
+        terminate_session()
+
+    except Exception as ex:
```

Comment:
Lame they don't have a custom exception to catch here :/. Looks like you could [check `flask.g`](https://github.com/puiterwijk/flask-oidc/blob/master/flask_oidc/__init__.py#L241-L242) if you want something slightly cleaner

Reply:
flask.g is hardly cleaner :)  i concur that it's lame.  this should never happen regardless, only found when debugging the bogus session, but leaving in tact as it does make it more robust.